zrml/global-disputes/src/lib.rs

```diff
@@ -0,0 +1,544 @@
+        }
+    }
+
+    impl<T> GlobalDisputesPalletApi<MarketIdOf<T>, AccountIdOf<T>, BalanceOf<T>> for Pallet<T>
+    where
+        T: Config,
+    {
+        fn push_voting_outcome(
+            market_id: &MarketIdOf<T>,
+            outcome: OutcomeReport,
+            owner: &T::AccountId,
+            vote_balance: BalanceOf<T>,
+        ) {
+            match <Winners<T>>::get(market_id) {
+                Some(winner_info) if winner_info.is_finished => return,
+                _ => (),
+            }
+            let update_winner = |b| {
+                <Winners<T>>::mutate(market_id, |highest: &mut Option<WinnerInfoOf<T>>| {
+                    *highest = Some(highest.clone().map_or(
+                        WinnerInfo::new(outcome.clone(), b),
+                        |prev_winner_info| {
+                            if b >= prev_winner_info.vote_sum {
+                                WinnerInfo::new(outcome.clone(), b)
+                            } else {
+                                prev_winner_info
+                            }
+                        },
+                    ));
+                });
+            };
+            match <Outcomes<T>>::get(market_id, &outcome) {
+                Some(mut outcome_info) => {
+                    let outcome_sum = outcome_info.outcome_sum.saturating_add(vote_balance);
+                    update_winner(outcome_sum);
+                    outcome_info.outcome_sum = outcome_sum;
+                    // there can not be more than MaxDisputes owners
+                    if outcome_info.owners.try_push(owner.clone()).is_ok() {
+                        <Outcomes<T>>::insert(market_id, outcome, outcome_info);
+                    } else {
+                        log::warn!(
+                            "Warning: The voting outcome was not added.  This happens because \
+                             there are too many voting outcome owners (length is {:?}).",
+                            &outcome_info.owners.len()
+                        );
+                    }
+                }
+                None => {
+                    // adding one item to BoundedVec can not fail
+                    if let Ok(owners) = BoundedVec::try_from(vec![owner.clone()]) {
+                        update_winner(vote_balance);
+                        let outcome_info = OutcomeInfo { outcome_sum: vote_balance, owners };
+                        <Outcomes<T>>::insert(market_id, outcome, outcome_info);
+                    }
+                }
+            }
+        }
+
+        fn get_voting_winner(market_id: &MarketIdOf<T>) -> Option<OutcomeReport> {
+            match <Winners<T>>::get(market_id) {
+                Some(mut winner_info) => {
+                    winner_info.is_finished = true;
+                    let winner_outcome = winner_info.outcome.clone();
+                    <Winners<T>>::insert(market_id, winner_info);
```

Comment:
I am confused, this is a `getter` function (I assume this because it starts with `get_`), and yet it applies changes like a setter functions.

Reply:
46fd6a7332f7bb0c66fd5fe25a24c07010d8d68b